src/twisted/newsfragments/11841.bugfix

```diff
@@ -0,0 +1 @@
+`defer.Deferred.fromCoroutine` now correctly processes pending futures.
```

Comment:
By reading the code, it looks like any `@defer.inlineCallbacks` is affected by this change.

I don't have any experience with futures and coroutines... so maybe "correctly" makes sense for someone who has experience.

But I think that it would help add more information about what is considered "correctly"

Reply:
Okey, I will explain it in more detail.